packages/dd-trace/src/id.js

```diff
@@ -44,11 +44,25 @@ function createBuffer (value) {
   if (value === '0') return zeroId
   if (!value) return pseudoRandom()
 
-  const size = Math.ceil(value.length / 2)
-  const buffer = new Uint8Array(size)
+  const buffer = new Uint8Array(8)
+
+  if (value.length < 16) {
+    value = Array(16 - value.length + 1).join('0') + value
+  }
 
-  for (let i = 0; i < size; i++) {
-    buffer[i] = parseInt(value.substring(i * 2, i * 2 + 2), 16)
+  if (value.length === 16) {
```

Comment:
couldn't these two branches be collapsed? the offset will be zero for length=16

Reply:
Sure. Actually it's even simpler than that, because after the padding, the length is always >= 16 so we don't need that if-block at all.

See latest push.